ytypes/unmarshal.go

```diff
@@ -29,6 +29,14 @@ type UnmarshalOpt interface {
 	IsUnmarshalOpt()
 }
 
+// BestEffortUnmarshal is an unmarshal option that 'ignores' errors while unmarshalling,
+// and continues the unmarshaling process. A failing unmarshal will still fail, but it will
+// fail as a series of error reports.
```

Comment:
What kind of error type will be returned?

One simple way of doing the error report is gnmi's errlist package: https://pkg.go.dev/github.com/openconfig/gnmi/errlist

However for more detailed error reporting you might consider what is being used in ygnmi:
https://github.com/openconfig/ygnmi/blob/613186339658c020a54b31ebe8e751e94e10ab43/ygnmi/unmarshal.go#L65-L74

In fact ygnmi does a very similar operation of unmarshalling Notifications to a GoStruct, and I might be able to leverage your effort here if you choose this route. I have been planning to use UnmarshalNotifications in ygnmi to do this.

Reply:
For this we've decided to return an error type that's to-be-filled.